pkg/apis/lbaas/v1/rule_genclient.go

```diff
@@ -0,0 +1,81 @@
+package v1
+
+import (
+	"context"
+	"net/url"
+
+	"go.anx.io/go-anxcloud/pkg/api/types"
+)
+
+// EndpointURL returns the URL where to retrieve objects of type Rule and the identifier of the given Rule.
+// It implements the api.Object interface on *Rule, making it usable with the generic API client.
+func (r *Rule) EndpointURL(ctx context.Context) (*url.URL, error) {
+	op, err := types.OperationFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	u, err := url.Parse("/api/LBaaS/v1/rule.json")
+	if err != nil {
+		return nil, err
+	}
+
+	if op == types.OperationList {
+		filters := r.buildListFilters()
+		query := u.Query()
+		query.Add("filters", filters.Encode())
+		u.RawQuery = query.Encode()
+	}
+
+	return u, nil
+}
+
+func (r *Rule) buildListFilters() url.Values {
```

Comment:
PR #114 has a new FilterHelper which could be of great help here, maybe we want to extract that from that PR and start using it

Reply:
pushed a commit to use the filter helper instead, please take a look at the changes @marioreggiori 

also added tests for the filters

Reply:
changes LGTM